application/org.openjdk.jmc.flightrecorder.graphview/src/main/java/org/openjdk/jmc/flightrecorder/ext/graphview/graph/DotGenerator.java

```diff
@@ -261,11 +261,23 @@ public String getColor(Edge edge) {
 	/**
 	 * Renders a {@link StacktraceGraphModel} in DOT format.
 	 */
-	public static String toDot(StacktraceGraphModel model, Map<ConfigurationKey, String> configuration) {
+	public static String toDot(
+		StacktraceGraphModel model, int maxNodesRendered, Map<ConfigurationKey, String> configuration) {
 		StringBuilder builder = new StringBuilder(2048);
 		String graphName = getConf(configuration, ConfigurationKey.Name, DEFAULT_NAME);
 		builder.append(String.format("digraph \"%s\" {%n", graphName));
-
+		int nodeCount = model.getNodes().size();
```

Comment:
I'm not sure I like the idea of special messages like this being passed as a "special" di-graph. I'd prefer this special check to be done outside of the toDot serializer, but I think it's okay since the graph view is sort of early access. It's certainly not worth holding up the release for.

Reply:
Agreed, this is a slight abuse of the serialiser 😃  but having it this way keeps the code simpler. We won't need to have special conditions before serialisation (we'd want to avoid if there are too many nodes), avoids adding extra behaviour in Java for passing messages and avoids adding functionality in the JS to do the toggle between displaying messages and showing the graph. Doing all this is not too difficult, but this way is simpler and doesn't introduce any new components.

We also do a similar thing for the flame graph: we display 'No stacktrace information available' as a flame graph with one root frame.

Related, doing the toggle with pure CSS (i.e. no JS) is not straightforward because graphviz resizes to cover the entire area. It might be possible to hack something in CSS that produces different display depending on the structure of the SVG generated - but given the state of the browser running inside our application, I don't think this is a good idea even if we get it working.